services/user/Branding/plugin/src/lib.rs

```diff
@@ -17,14 +17,31 @@ use errors::ErrorType;
 
 struct BrandingPlugin;
 
+struct Latch;
+
+impl Latch {
+    fn new() -> Self {
+        set_propose_latch(Some("branding")).unwrap();
+        Self
+    }
+}
+
+impl Drop for Latch {
+    fn drop(&mut self) {
+        set_propose_latch(None).unwrap();
+    }
+}
+
 impl Api for BrandingPlugin {
     fn set_network_name(name: String) {
+        let _latch = Latch::new();
+
         let packed_network_name_args = branding::action_structs::setNetworkName { name }.packed();
-        set_propose_latch(Some("branding")).unwrap();
         add_action_to_transaction("setNetworkName", &packed_network_name_args).unwrap();
     }
     fn set_logo(logo: Vec<u8>) {
-        set_propose_latch(Some("branding")).unwrap();
+        let _ = Latch::new();
```

Comment:
Is this right? I thought _ doesn't bind anything nor does it extend the lifetime of the initializer.

Reply:
You're correct, I noticed that and fixed it for `set_network_name` and forgot to update it for `set_logo`.
Fixed